evm/src/cpu/kernel/asm/main.asm

```diff
@@ -14,14 +14,18 @@ global main:
     // Initialise the shift table
     %shift_table_init
 
-    // Second, load all MPT data from the prover.
-    PUSH hash_initial_tries
-    %jump(load_all_mpts)
-
 global hash_initial_tries:
-    %mpt_hash_state_trie   %mload_global_metadata(@GLOBAL_METADATA_STATE_TRIE_DIGEST_BEFORE)    %assert_eq
+    // We compute the length of the trie data segment in `mpt_hash` so that we
+    // can check the value provided by the prover.
+    PUSH 1 // Initial length
```

Comment:
nit: maybe emphasize why the segment is initially of length 1 (i.e. containing 0) for clarity?